### PR TITLE
Cobertura de teste do DesignSystemColors

### PR DESCRIPTION
**Pull request by shonorio**
## Objetivo

Escrever testes para o DesignSystemColors e atingir a meta de cobertura para esta classe

## Execução

1. Criado os testes para o DesignSystemColors
2. Desativado coverage onde não é possível testar

### CODE REVIEW

**Review by pedrox-hs — DISMISSED** [review]

**Review by diraol — COMMENTED** [review]
@shonorio can we add one or two tests that fail the conversion? To test the "negative case"?

**Comment by shonorio** [comment]
@diraol e qual é o comportamento para o "teste negativo"? Hoje o código não tem resultado para isto, estão precisa definir um comportamento para a falha de conversão. Vai mudar a assinatura para devolver nulo? Se fizer isto, vai ter um impacto em quem já está usando.

**Comment by shonorio** [comment]
@diraol @pedroque  Quando forem adicionar este código a master, podem fazer por `merge` e não `squash`? Pq? É que o `squash` perde o históricos dos `commits`. Estou tomando o cuidado de realizar `commits` documentais e que pode ajudar a entender a evolução do código no futuro.

**Comment by diraol** [comment]
> @diraol e qual é o comportamento para o "teste negativo"? Hoje o código não tem resultado para isto, estão precisa definir um comportamento para a falha de conversão. Vai mudar a assinatura para devolver nulo? Se fizer isto, vai ter um impacto em quem já está usando.

A minha pergunta é, o que vai acontecer se alguém chamar esse método com `ZZZZZZ` ou com `XZK#@!$%%`? Ou mesmo "123456123456"...

**Comment by shonorio** [comment]
> > @diraol e qual é o comportamento para o "teste negativo"? Hoje o código não tem resultado para isto, estão precisa definir um comportamento para a falha de conversão. Vai mudar a assinatura para devolver nulo? Se fizer isto, vai ter um impacto em quem já está usando.
> 
> A minha pergunta é, o que vai acontecer se alguém chamar esse método com `ZZZZZZ` ou com `XZK#@!$%%`? Ou mesmo "123456123456"...

Vai dar crash, não tem tratamento para este tipo de input. O `int.parse(hexColor, radix: 16)` não vai processar esta informação. A pergunta continua, que tipo de comportamento é desejado nesta situação? Retornar nulo? Retornar um valor default? Ou simplesmente considera isto nos testes, tipo 'valor inválido crash'?

**Comment by pedrox-hs** [comment]
@shonorio acredito que o que o @diraol se refere, é em testar se realmente está lançando a exception caso a string seja invalida, mesmo que sabendo que se isso realmente acontecer, irá causar um crash, mas assim evitaria caso acidentalmente a implementação mude e acabe permitindo valores invãlidos.

(ps. vc está marcando o outro Pedro rs)

**Comment by shonorio** [comment]
@diraol cobertura de teste atualizado para incluir o cenário de Exception por hex inválido.

File: test/app/shared/design_system/colors_test.dart [hunk 1]
```diff
@@ -21,5 +21,10 @@ void main() {
       expect(DesignSystemColors.hexColor(hexColor), expectedColor);
       expect(DesignSystemColors.hexColor(hexColorWithoutHash), expectedColor);
     });
+
+    test('return a Exceptions for an invalid hexColor', () {
```
Comment (diraol) [inline test/app/shared/design_system/colors_test.dart:25]:
```suggestion
    test('return Exception for an invalid hexColor', () {
```

**Review by diraol — APPROVED** [review]

**Review by pedrox-hs — APPROVED** [review]
